.github/workflows/releases.yml

```diff
@@ -0,0 +1,149 @@
+name: Releases
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  memcoin:
+    name: release memcoin and .deb
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Use go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '>=1.18'
+
+      - name: Install fpm
+        run: |
+          sudo apt-get update
+          sudo apt-get install ruby-dev build-essential
+          sudo gem install fpm -f
+
+      - name: build artifacts
+        # builds the binary and the .deb
+        run: make deb
+
+      - name: Publish release to aptly
+        env:
+          APTLY_USER: ${{ secrets.APTLY_USER }}
+          APTLY_PASSWORD: ${{ secrets.APTLY_PASSWORD }}
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+        run: |
+          ./deb-package/upload-artifacts.sh deb-package/dist
+
+      - name: Update artifacts to Github's release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            memcoin-*
+            deb-package/dist/*
+
+  backend:
+    name: release the web backend
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: ./web/backend
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Get the version
+        id: get_version
+        run: |
+          echo ::set-output name=version::$(echo ${GITHUB_REF/refs\/tags\//})
+          echo ::set-output name=version_file::web-backend-$(echo ${GITHUB_REF/refs\/tags\//} | tr . _)
+          echo "::set-output name=shortsha::$(git rev-parse --short ${GITHUB_SHA})"
+          echo "::set-output name=buildurl::${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/commit/${GITHUB_SHA}"
+          echo "::set-output name=date::$(date +'%d/%m/%y %H:%M')"
```

Comment:
I'm very confused by this block of code 🤔

Reply:
it declares variables that are subsequently available for the next steps, using the step id, e.g. `{{ steps.get_version.version }}`.